content/en/blog/_posts/2022-08-29-csi-inline-volumes-ga.md

```diff
@@ -0,0 +1,98 @@
+---
+layout: blog
+title: "Kubernetes 1.25: CSI Inline Volumes have graduated to GA"
+date: 2022-08-29
+slug: csi-inline-volumes-ga
+---
+
+**Author:** Jonathan Dobson (Red Hat)
+
+CSI Inline Volumes were introduced as an alpha feature in Kubernetes 1.15 and have been beta since 1.16. We are happy to announce that this feature has graduated to General Availability (GA) status in Kubernetes 1.25.
+
+CSI Inline Volumes are similar to other ephemeral volume types, such as `configMap`, `downwardAPI` and `secret`. The important difference is that the storage is provided by a CSI driver, which allows the use of ephemeral storage provided by third-party vendors. The volume is defined as part of the pod spec and follows the lifecycle of the pod, meaning the volume is created once the pod is scheduled and destroyed when the pod is destroyed.
+
+## What's new in 1.25?
+
+There are a couple of new bug fixes related to this feature in 1.25, and the [CSIInlineVolume feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) has been locked to `True` with the graduation to GA. There are no new API changes, so users of this feature during beta should not notice any significant changes aside from these bug fixes.
+
+- [#89290 - CSI inline volumes should support fsGroup](https://github.com/kubernetes/kubernetes/issues/89290)
+- [#79980 - CSI volume reconstruction does not work for ephemeral volumes](https://github.com/kubernetes/kubernetes/issues/79980)
+
+## When to use this feature
+
+CSI inline volumes are meant for simple local volumes that should follow the lifecycle of the pod. They may be useful for providing secrets, configuration data, or other special-purpose storage to the pod from a CSI driver.
+
+A CSI driver is not suitable for inline use when:
+- The volume needs to persist longer than the lifecycle of a pod
+- Volume snapshots, cloning, or volume expansion are required
+- The CSI driver requires `volumeAttributes` that should be restricted to an administrator
+
+## How to use this feature
```

Comment:
Add a link to secrete store CSI driver as an example implementation?  https://github.com/kubernetes-sigs/secrets-store-csi-driver#features

Reply:
Good idea, I added a new section with example use cases.

Reply:
A couple of examples:
Using a container image to ship data rather then programs: https://github.com/kubernetes-csi/csi-driver-image-populator/blob/master/examples/pod.yaml
Support safe privileged/unprivilaged communication https://github.com/spiffe/spiffe-csi/blob/main/example/config/workload.yaml